compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7097,7 +7097,9 @@ public void visit(BLangTypeConversionExpr conversionExpr) {
     @Override
     public void visit(BLangLambdaFunction bLangLambdaFunction) {
         // Collect all the lambda functions.
-        env.enclPkg.lambdaFunctions.add(bLangLambdaFunction);
+        if (!env.enclPkg.lambdaFunctions.contains(bLangLambdaFunction)) {
```

Comment:
Added the lambda functions in the query and the lambda functions we create for the clauses when we are visiting the lambda functions in `QueryDesugar` [1]. This was done to ensure that they are added in the order they are visited so that the closures are resolved properly. Hence those should not be added to the `env.enclPkg.lambdaFunctions` when they are being visited in `Desugar`.
[1] https://github.com/ballerina-platform/ballerina-lang/blob/740c7ec80a28b255773fa79052418996b73b96b9/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java#L1329